packages/utils/src/ratelimit.ts

```diff
@@ -43,9 +43,14 @@ export function isRateLimited(limits: RateLimits, category: string, now: number
  */
 export function updateRateLimits(
   limits: RateLimits,
-  headers: Record<string, string | null | undefined>,
+  newRateLimitParameters: {
```

Comment:
Let's type this to be `TransportMakeRequestResponse`? Then we can just do a check against headers with an early return, otherwise we know the headers should be defined.

Reply:
I don't quite get what you mean with the early return.

Do you mean it like this:

```ts
if (headers?.['x-sentry-rate-limits'] && headers?.['retry-after'] && statusCode === 429) {
    updatedRateLimits.all = now + 60 * 1000;
    return;
}
```

Not sure if that is cleaner...

Reply:
You know what - ignore me on the headers check. I guess the header items can be null anyway.

I would still use the `TransportMakeRequestResponse` type to make things consistent though.